autometa/common/kmers.py

```diff
@@ -82,11 +82,13 @@ def init_kmers(kmer_size=5):
         {kmer:index, ...}
 
     """
-    kmers = {}
+    if not float(kmer_size).is_integer():
+        raise TypeError(f"kmer_size must be an int! Given: {type(kmer_size)}")
+    kmer_size = int(float(kmer_size))
```

Comment:
Let's simplify this and not try any conversion. Seems like a fair enough ask from the user.

Also, if we wanted, we could start to use [type hints](https://docs.python.org/3/library/typing.html "type hints python page") as well.

```suggestion
    if not isinstance(kmer_size, int):
        raise TypeError(f"kmer_size must be an int! Given: {type(kmer_size)}")
```